### PR TITLE
refactor(documentations): pluck cover_path in controller

### PR DESCRIPTION
**Pull request by semia**
- Instead of request Documentation on the view use a method pluck cover_path in the controller

- Add pluck_cover_path_documentation method in documentations controler and add it in before_action
- Delete all.pluck on questions controller replace by pluck
- Add pluck_name_id_documentation method  in questions controller and add it in before_action

Related to issue #127

### CODE REVIEW

File: app/controllers/admin/documentations_controller.rb [hunk 1]
```diff
@@ -49,4 +50,8 @@ def documentation_params
   def assign_documentation
     @documentation = Documentation.find(params[:id])
   end
+
+  def pluck_cover_path_documentation
+    @documentations = Documentation.pluck(:cover_path)
```
Comment (arnlen) [inline app/controllers/admin/documentations_controller.rb]: Nice!
To be more precise - and not confusing, especially when into the view - I recommend you to rename this `@documentations_cover_path`

File: app/controllers/admin/documentations_controller.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class Admin::DocumentationsController < AdminController
   before_action :assign_documentation, only: %i(show edit update destroy)
+  before_action :pluck_cover_path_documentation, only: %i(new edit)
```
Comment (arnlen) [inline app/controllers/admin/documentations_controller.rb]: Good idea! 👍 
To stay coherent with the above before_action, you could rename this method `assign_documentations_cover_path`.

File: app/controllers/questions_controller.rb [hunk 3]
```diff
@@ -1,5 +1,6 @@
 class QuestionsController < ApplicationController
   before_action :assign_question, only: %i(show edit update destroy)
+  before_action :pluck_name_id_documentation, only: %i(new edit)
```
Comment (arnlen) [inline app/controllers/questions_controller.rb]: Same as above: a better naming could improve the comprehension.
Better it's an "action", starting the method name by an "action verb" would help a lot: `assign_documentations_name_id`.

File: app/controllers/questions_controller.rb [hunk 4]
```diff
@@ -9,7 +10,6 @@ def show
   end
 
   def new
-    @documentations = Documentation.all.pluck(:name, :id)
```
Comment (arnlen) [inline app/controllers/questions_controller.rb]: 👍

File: app/controllers/questions_controller.rb [hunk 5]
```diff
@@ -24,7 +24,6 @@ def create
   end
 
   def edit
-    @documentations = Documentation.all.pluck(:name, :id)
```
Comment (arnlen) [inline app/controllers/questions_controller.rb]: 👍

File: app/views/admin/documentations/_form.html.haml [hunk 6]
```diff
@@ -7,5 +7,5 @@
   = f.text_field :short_name, class: 'form-control', id: 'Abbréviation', required: true
 .field
   %h4= f.label 'Couverture'
-  = f.select(:cover_path, options_for_select(Documentation.all.map { |d| [d.cover_path] }), {}, {class: 'form form-group form-control'})
```
Comment (arnlen) [inline app/views/admin/documentations/_form.html.haml]: 😱 

lol!
Nice fix! 😆

File: app/views/admin/documentations/_form.html.haml [hunk 7]
```diff
@@ -7,5 +7,5 @@
...
+  = f.select(:cover_path, options_for_select(@documentations), {}, {class: 'form form-group form-control'})
```
Comment (arnlen) [inline app/views/admin/documentations/_form.html.haml]: 😍

**Comment by semia** [comment]
Oki thx :+1:

**Comment by arnlen** [comment]
Perfect, thx, I merge. 😃

**Pull request merged**